.github/workflows/ci.yml

```diff
@@ -0,0 +1,96 @@
+      - name: lint modules/azure folder
+        id: azure_module_lint
+        uses: golangci/golangci-lint-action@v2.2.0
+        with:
+          version: v1.30
+          working-directory: modules/azure
+      - name: list test files to lint
+        run: |
+          [ -d "test/azure" ] && ls -li "test/azure"
+      - name: lint test/azure folder
+        id: azure_test_lint
+        uses: golangci/golangci-lint-action@v2.2.0
+        with:
+          version: v1.30
+          working-directory: test/azure
+      - name: login to azure cli
+        uses: azure/login@v1.1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+      - name: run go test for azure
+        id: azure_test
+        env:
+          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+        run: |
+          az account show
+          cd test
+          TEST_VERSION=`date +"%Y%m%d%H%M"`
+          export ARM_SUBSCRIPTION_ID=`az account show --query "id" --output tsv`
+
+          APP_ID=`echo $AZURE_CREDENTIALS | jq -r -c ".clientId"`
+
+          APP_PASSWORD=`echo $AZURE_CREDENTIALS | jq -r -c ".clientSecret"`
+
+          TENANT_ID=`echo $AZURE_CREDENTIALS | jq -r -c ".tenantId"`
+
+          export ARM_CLIENT_ID="$APP_ID"
+          export ARM_CLIENT_SECRET="$APP_PASSWORD"
+          export ARM_SUBSCRIPTION_ID="$ARM_SUBSCRIPTION_ID"
+          export ARM_TENANT_ID="$TENANT_ID"
+
+          rm -rf ssh_key*
+          ssh-keygen -m PEM -t rsa -b 4096 -f ./ssh_key -q -N ""
+          export TF_VAR_ssh_public_key="$PWD/ssh_key.pub"
+          export TF_VAR_client_id="$APP_ID"
+          export TF_VAR_client_secret="$APP_PASSWORD"
+
+          go test ./azure/* -timeout 90m
+      - name: report back the result
+        if: always()
+        env:
+          CURRENT_REPOSITORY: ${{ github.repository }}
+          RUN_ID: ${{ github.run_id }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          TARGET_REPOSITORY: ${{ github.event.inputs.target_repository }}
+          TARGET_PR: ${{ github.event.inputs.target_pr }}
+          TEST_RESULT: ${{ steps.azure_test.conclusion }}
+          TEST_LINT_RESULT: ${{ steps.azure_test_lint.conclusion }}
+          MODULE_LINT_RESULT: ${{ steps.azure_module_lint.conclusion }}
+        run: |
+          if [ "$TEST_RESULT" == "success" ] && [ "$TEST_LINT_RESULT" == "success" ] && [ "$MODULE_LINT_RESULT" == "success" ]; then
+            BODY_PAYLOAD="[Microsoft CI Bot] TL;DR; success :thumbsup:\n\nYou can check the status of the CI Pipeline logs here ; https://github.com/${CURRENT_REPOSITORY}/actions/runs/$RUN_ID"
+            curl --request POST --header "Authorization: token ${GITHUB_TOKEN}" --header "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${TARGET_REPOSITORY}/issues/${TARGET_PR}/comments --data "{\"body\":\"${BODY_PAYLOAD}\"}"
+          elif [ "$TEST_RESULT" != "failure" ] || [ "$TEST_LINT_RESULT" != "failure" ] || [ "$MODULE_LINT_RESULT" != "failure" ]; then
```

Comment:
Why an `elif` here rather than just `else`?

Reply:
we're checking the previous steps statuses, they can be "success", "failure", "skipped" ,etc.

I want to check if all the steps are success or at least 1 step failed.

I need to check 2 specific condition, that's why I can't use `else`.

Reply:
Apologies if I'm missing something obvious, but it seems like the logic here is:

1. If all the checks were successful, update the PR with a success message.
1. Else if at least one of the checks was NOT failure, then update the PR with a failure message. 

I'm not sure I understand the second branch? Shouldn't we update the PR with failure if any of the checks were failures? Or even simpler, shouldn't it just be an "else"? As in, if all the checks weren't successful (the if-statement), then there must have been a failure, so update the PR accordingly?

Reply:
what was I thinking when I wrote that line? 🤦‍♂️ I even thought I tested that clause (https://github.com/polatengin/terratest/pull/14#issuecomment-685994863)

it should be "if test step failed OR lint test folder step failed OR lint module folder step failed"

pushed the fix and tested with cases, it should be ok now, sorry for the confusion...

Reply:
OK, that new version makes much more sense 😄 

One more sanity check on these if-statements: what are the other possible statuses? Right now, it updates the PR with (a) a success message if everything succeeded or (b) a failure message if there were any failures. Should it update the PR on other statuses? E.g., If tests were skipped, would that be useful to know?

Reply:
conclusion of a step can be, `failure`, `success`, `cancelled` or `skipped`

as you already mentioned, we're creating a comment on either "all success" or "any failure" conditions.

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context